api/types/container/secret.go

```diff
@@ -0,0 +1,14 @@
+package container
+
+import "os"
+
+// ContainerSecret represents a secret in a container.  This gets realized
+// in the container tmpfs
+type ContainerSecret struct {
```

Comment:
Again, this repeats secretreference. Why aren't we using that type for what it is intended?